CONTRIBUTING.md

```diff
@@ -55,55 +60,144 @@ Prerequisites:
 Steps:
 
 - Clone repository
-- Open it in editor
+- Open it in your editor
 - For VS Code, execute `Dev Containers: Reopen in container` command
 
 For detailed instructions, please see the [Dev Containers tutorial](https://code.visualstudio.com/docs/devcontainers/tutorial).
 
-### Extra tidbits
+### 🧱 Structure of the Repository
+
+If you're a newcomer to Gradio, we recommend getting familiar with the overall structure of the repository so that you can focus on the part of the source code you'd like to contribute to.
 
-- You can run gradio scripts in reload mode which will watch for changes in the `gradio` folder and reload the app if changes are made.
+- `/gradio`: contains the Python source code for the library
+  - `/gradio/interface.py`: contains the Python source code for the core `Interface` class
+  - `/gradio/blocks.py`: contains the Python source code for the core `Blocks` class
+  - `/gradio/components.py`: contains the Python source code for the `components`, you can add your custom components here.
+- `/js`: contains the HTML/JS/CSS source code for the library ([start here for frontend changes](/js/README.md))
+- `/test`: contains Python unit tests for the library
+- `/demo`: contains demos that are used in the documentation, you can find `Gradio` examples over here.
+- `/website`: contains the code for the Gradio website (www.gradio.app). See the README in the `/website` folder for more details
+
+
+
+- To develop the frontend app, you should also follow [js/README.md](js/README.md).
```

Comment:
Maybe we should move the relevant sections to CONTRIBUTING.md? Its easier on users if they don't have to go back and forth two readmes